aiida/manage/external/postgres.py

```diff
@@ -128,28 +151,70 @@ def drop_db(self, dbname):
         """
         Drop a database in postgres
 
-        :param dbname: (str), Name of the database.
+        :param str dbname: Name of the database.
         """
         self.execute(_DROP_DB_COMMAND.format(dbname))
 
     def copy_db(self, src_db, dest_db, dbuser):
         self.execute(_COPY_DB_COMMAND.format(dest_db, src_db, dbuser))
 
-    def db_exists(self, dbname):
+    def check_db(self, dbname):
+        """Looks up if a database with the name exists, prompts for using or creating a differently named one.
+
+        :param str dbname: Name of the database to be created or reused.
+        :returns: tuple (dbname, created)
         """
-        Check wether a postgres database with dbname exists
+        create = True
+        while create and self.db_exists(dbname):
+            echo.echo_info('database {} already exists!'.format(dbname))
+            if not click.confirm('Use it (make sure it is not used by another profile)?'):
+                dbname = click.prompt('new name', type=str, default=dbname)
```

Comment:
I realize now (and not sure if this was already the case) that the code here is actually tightly coupled to the CLI. Is this code _only_ used from the CLI, or should it be? Maybe not for this PR, but it would be good to think about this and see if we can make this code more general or really just move it to `aiida.cmdline` if it really is just meant for the CLI

Reply:
Yeah, there are many pieces like this in the code, where certain functions can be called in "interactive" mode and then start prompting.

Once we have a clear path forward how to deal with those, we should go through the entire codebase once and move them.